pkg/cache/cluster_test.go

```diff
@@ -189,41 +190,80 @@ func Benchmark_sync(t *testing.B) {
 	}
 }
 
+type AlwaysEnabledGates struct{}
+
+func (AlwaysEnabledGates) Enabled(clientfeatures.Feature) bool {
+	return true
+}
+
 func TestEnsureSynced(t *testing.T) {
-	obj1 := &appsv1.Deployment{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "apps/v1",
-			Kind:       "Deployment",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "helm-guestbook1",
-			Namespace: "default1",
-		},
-	}
-	obj2 := &appsv1.Deployment{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "apps/v1",
-			Kind:       "Deployment",
+	tests := []struct {
+		name               string
+		listUsingWatchAPIs bool
+	}{
+		{
+			name:               "WatchAPIUsed",
+			listUsingWatchAPIs: true,
 		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "helm-guestbook2",
-			Namespace: "default2",
+		{
+			name:               "WatchAPINotUsed",
+			listUsingWatchAPIs: false,
 		},
 	}
 
-	cluster := newCluster(t, obj1, obj2)
-	err := cluster.EnsureSynced()
-	require.NoError(t, err)
+	originalFeatureGates := clientfeatures.FeatureGates()
 
-	cluster.lock.Lock()
-	defer cluster.lock.Unlock()
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.listUsingWatchAPIs {
+				// Enable WatchListClient in particular. Setting via env variable here is too late.
+				clientfeatures.ReplaceFeatureGates(AlwaysEnabledGates{})
```

Comment:
Looks like this enables every feature gate that can be enabled. Can we instead enable just the watch-list gate?

Reply:
It's a bit more complicated, and it's just for this test, so I'd rather keep this.

Reply:
Although shouldn't be too hard to construct gates more precisely.